packages/db/tests/translate/test-pdr.js

```diff
@@ -0,0 +1,226 @@
+
+  const fakeExecution = { arn: 'arn:aws:execution:1234abcd' };
+  const fakeExecutionPgModel = {
+    get: () => Promise.resolve(fakeExecution),
+  };
+
+  const fakeProvider = { name: 'ABCprov' };
+  const fakeProviderPgModel = {
+    get: () => Promise.resolve(fakeProvider),
+  };
+
+  const postgresPdr = {
+    status: 'completed',
+    name: 'acbd1234.PDR',
+    collection_cumulus_id: 1,
+    provider_cumulus_id: 1,
+  };
+
+  const expectedPdr = {
+    pdrName: postgresPdr.name,
+    provider: fakeProvider.name,
+    collectionId: constructCollectionId(fakeCollection.name, fakeCollection.version),
+    status: postgresPdr.status,
+  };
+
+  const translatedPdrRecord = await translatePostgresPdrToApiPdr(
+    postgresPdr,
+    {},
+    fakeCollectionPgModel,
+    fakeProviderPgModel,
+    fakeExecutionPgModel
+  );
+
+  t.deepEqual(
+    translatedPdrRecord,
+    expectedPdr
+  );
+});
+
+test('translateApiPdrToPostgresPdr converts API PDR to Postgres', async (t) => {
+  const record = {
+    pdrName: 'name',
+    status: 'running',
+    provider: 'fake-provider',
+    collectionId: 'fake-collection___000',
+    progress: 50,
+    stats: {
+      running: ['arn3', 'arn4'],
+      completed: ['arn1', 'arn2'],
+    },
+    address: 'address',
+    PANSent: true,
+    PANmessage: 'message',
+    originalUrl: 'url',
+    execution: 'execution',
+    timestamp: Date.now(),
+    duration: 10000,
+    createdAt: Date.now(),
+    updatedAt: Date.now(),
+  };
+
+  const fakeKnex = {};
+  const fakeCollectionPgModel = {
+    getRecordCumulusId: () => Promise.resolve(1),
+  };
+  const fakeExecutionPgModel = {
+    getRecordCumulusId: () => Promise.resolve(2),
+  };
+  const fakeProviderPgModel = {
+    getRecordCumulusId: () => Promise.resolve(3),
+  };
+
+  const expectedPostgresPdr = {
+    name: record.pdrName,
+    status: record.status,
+    address: record.address,
+    progress: record.progress,
+    pan_sent: record.PANSent,
+    pan_message: record.PANmessage,
+    original_url: record.originalUrl,
+    duration: record.duration,
+    stats: record.stats,
+    created_at: new Date(record.createdAt),
+    updated_at: new Date(record.updatedAt),
+    timestamp: new Date(record.timestamp),
+    collection_cumulus_id: 1,
+    execution_cumulus_id: 2,
+    provider_cumulus_id: 3,
+  };
+
+  const result = await translateApiPdrToPostgresPdr(
+    record,
+    fakeKnex,
+    fakeCollectionPgModel,
+    fakeProviderPgModel,
+    fakeExecutionPgModel
+  );
+  t.deepEqual(
+    result,
+    expectedPostgresPdr
+  );
+});
+
+test('translateApiPdrToPostgresPdr handles optional fields', async (t) => {
+  const record = {
+    pdrName: 'name',
+    status: 'running',
+    provider: 'fake-provider',
+    collectionId: 'fake-collection___000',
+  };
+
+  const fakeKnex = {};
+  const fakeCollectionPgModel = {
+    getRecordCumulusId: () => Promise.resolve(1),
+  };
+  const fakeExecutionPgModel = {
+    getRecordCumulusId: () => Promise.resolve(2),
+  };
```

Comment:
Same here: we don't need this fake model for this test

Reply:
https://github.com/nasa/cumulus/pull/2273/commits/7ed58e6a1895c91c2d21825e119ebfa9ace452bc